addons/product/models/product_template.py

```diff
@@ -538,7 +538,29 @@ def copy(self, default=None):
             default = {}
         if 'name' not in default:
             default['name'] = _("%s (copy)", self.name)
-        return super(ProductTemplate, self).copy(default=default)
+
+        res = super(ProductTemplate, self).copy(default=default)
+
+        # Since we don't copy the attribute lines directly, we need to match the extra price
+        # and create the exclusions of the newly created attribute line values.
+        # This matching expects these values to be uniquely determined by the attribute
+        # value. If two lines uses the same attribute value, it will not work properly.
+        copied_ptavs = res.attribute_line_ids.product_template_value_ids
+        for ptav in self.attribute_line_ids.product_template_value_ids:
+            if ptav.price_extra or ptav.exclude_for:
+                copied_ptavs.filtered(
+                    lambda x: x.product_attribute_value_id == ptav.product_attribute_value_id
+                ).write({
+                    'price_extra': ptav.price_extra,
+                    'exclude_for': [Command.create({
+                        'product_tmpl_id': res.id,
+                        'value_ids': copied_ptavs.filtered(
+                            lambda x: x.product_attribute_value_id in excl.value_ids.product_attribute_value_id),
+                    }) if excl.product_tmpl_id == self else Command.link(excl.id)
```

Comment:
`Command.link(excl.id)` is wrong, this will remove the exclusion from the original product, to move it to the new one.

You will have to duplicate the exclusion, not move it from one ptav(product) to another.

Reply:
Thank you, I did not notice !